### PR TITLE
Add BranchProtectionConfigurationEvent and SecretScanningAlertLocationEvent

### PR DESCRIPTION
**Pull request by maditya**

### CODE REVIEW

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @maditya.
To add new events, we need to make some more changes to the following files:
* event_types_test.go
* messages.go
* messages_test.go

Please see #3258 for an example of the changes that are needed.
Thanks again!

**Comment by maditya** [comment]
```
--- FAIL: TestProjectsService_DeleteProjectCard (0.00s)
    projects_test.go:611: Projects.DeleteProjectCard returned error: Delete "http://127.0.0.1:39961/api-v3/projects/columns/cards/1": net/http: HTTP/1.x transport connection broken: http: CloseIdleConnections called
```
@gmlewis this seems like an intermittent failure unrelated to the changes in this PR. Unable to reproduce it locally. Can we re-run the workflow?

File: github/orgs_codesecurity_configurations_test.go
Comment (gmlewis) [inline github/orgs_codesecurity_configurations_test.go:385]: The changes to this file seem wrong to me.
I think you can revert the changes to this file.

Comment (maditya) [reply github/orgs_codesecurity_configurations_test.go:385]: This was a warning which I fixed.

Per https://pkg.go.dev/net/http 

```
If WriteHeader is not called explicitly, the first call to Write
will trigger an implicit WriteHeader(http.StatusOK).
```

So I think we should either keep this change or explicitly call `w.WriteHeader(http.StatusOK)` before we write to `w`. Any thoughts?

Comment (gmlewis) [reply github/orgs_codesecurity_configurations_test.go:385]: Oh, interesting, so this is unrelated to this PR.  OK.  I'm fine to keep the change.  Thanks.

**Review by gmlewis — APPROVED** [review]
Thank you, @maditya !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by tomfeigin — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @tomfeigin !
Merging.

**Pull request merged**
